pkg/apis/controlplane/v1beta2/types.go

```diff
@@ -293,3 +293,13 @@ type NetworkPolicyNodeStatus struct {
 	// The generation realized by the Node.
 	Generation int64 `json:"generation,omitempty" protobuf:"varint,2,opt,name=generation"`
 }
+
+// A PortMask is a representation of a single port name or number or a range of ports from base port with a bitmask applied.
+type PortMask struct {
```

Comment:
Maybe I missed something, but are we able to convert PortRange to PortMask? But how we express Except in PortRange?

Reply:
We use a third_party lib https://github.com/digitalocean/go-openvswitch/blob/master/ovs/portrange.go to do the conversion. 
Do you mean Except in PortMask? We will split PortRange into many ranges based on the Except, then use third party lib to convert a range to several ports with masks.